CMakeLists.txt

```diff
@@ -30,7 +34,10 @@ include(CMakePrintHelpers)
 include(F2Py)
 find_package(pybind11 CONFIG)
 
-find_package(Python REQUIRED COMPONENTS Interpreter Development)
+
+set(Python3_FIND_VIRTUALENV FIRST)
+find_package(Python COMPONENTS Interpreter Development)
```

Comment:
```suggestion
find_package(Python REQUIRED COMPONENTS Interpreter Development.Module)
```

Reply:
Hi @henryiii, thank you for suggestion. It works as well. 

By the way, there is also another problem, which probably you met somewhere. Now extension modules get a wrong suffix. For example, when the wheels build for python3.10, I get the following:

```
copying build/lib.macosx-10.9-x86_64-cpython-310/chromo/models/_dpmjetIII191.cpython-311-darwin.so -> build/bdist.macosx-10.9-x86_64/wheel/chromo/models
```
, i.e. `.cpython-311-darwin.so` is used instead of `.cpython-310-darwin.so`

I guess it is because CMake finds the wrong python version, because in

```
Found PythonInterp: /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibw-run-4x_bfv58/cp310-macosx_x86_64/build/venv/bin/python (found suitable version "3.10.8", minimum required is "3.6")
2023-02-10T14:19:59.9278720Z   -- Found PythonLibs: /Library/Frameworks/Python.framework/Versions/3.10/lib/libpython3.10.dylib
2023-02-10T14:19:59.9313940Z   -- Performing Test HAS_FLTO
2023-02-10T14:20:00.0537060Z   -- Performing Test HAS_FLTO - Failed
2023-02-10T14:20:00.1085600Z   -- Could NOT find pybind11 (missing: pybind11_DIR)
2023-02-10T14:20:02.5305190Z   -- Found Python: /usr/local/Frameworks/Python.framework/Versions/3.11/bin/python3.11 (found version "3.11.1") found components: Interpreter Development Development.Module Development.Embed
2023-02-10T14:20:10.0779340Z   -- CMAKE_BUILD_TYPE="Release"
2023-02-10T14:20:10.0804230Z   -- PYTHON_EXECUTABLE="/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibw-run-4x_bfv58/cp310-macosx_x86_64/build/venv/bin/python"
2023-02-10T14:20:10.0857870Z   -- PYTHON_INCLUDE_DIRS="/Library/Frameworks/Python.framework/Versions/3.10/include/python3.10"
2023-02-10T14:20:10.0862690Z   -- PYTHON_LIBRARIES="/Library/Frameworks/Python.framework/Versions/3.10/lib/libpython3.10.dylib"
2023-02-10T14:20:10.0870370Z   -- F2PY_INCLUDE_DIR="/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-build-env-6jgq5s24/overlay/lib/python3.10/site-packages/numpy/f2py/src
```

the line 

```
-- Found Python: /usr/local/Frameworks/Python.framework/Versions/3.11/bin/python3.11 (found version "3.11.1") found components: Interpreter Development Development.Module Development.Embed
```

points to the wrong path.

The problem is that it is not possible to know beforehand the path to the correct version of python, if it possible somehow to give a hint to CMake. So the paths should be found at runtime.

Reply:
The solution how give a hint is to set environment variable VIRTUAL_ENV which points to venv directory created by cibuildwheel. Usually VIRTUAL_ENV is created when `source ./venv/bin/activate` is run. But cibuildwheel doesn't do that. It is also difficult to set VIRTUAL_ENV using variables provided by cibuildwheel so that CMake could find it.  I set the environment variable VIRTUAL_ENV via python in `setup.py`.